src/parser/utils.ts

```diff
@@ -8,11 +10,20 @@ export function normalizedFunctionString(fn: Function): string {
   return fn.toString().replace('function(', 'function (');
 }
 
-function insecureRandomBytes(size: number): Uint8Array {
+const isReactNative =
+  typeof global.navigator === 'object' && global.navigator.product === 'ReactNative';
+
+const insecureWarning = isReactNative
+  ? 'BSON: For React Native please polyfill crypto.getRandomValues, e.g. using: https://www.npmjs.com/package/react-native-get-random-values.'
+  : 'BSON: No cryptographic implementation for random bytes present, falling back to a less secure implementation.';
+
+const insecureRandomBytes: RandomBytesFunction = function insecureRandomBytes(size: number) {
+  console.warn(insecureWarning);
```

Comment:
I think it'd be better to use [`process.emitWarning`](https://nodejs.org/api/process.html#process_process_emitwarning_warning_options) here.

```suggestion
  process.emitWarning(insecureWarning);
```

Reply:
Thanks, but wouldn't this throw in a browser context without crypto?